sdv/constraints/tabular.py

```diff
@@ -46,16 +46,60 @@ class CustomConstraint(Constraint):
             Function to replace the ``is_valid`` method.
     """
 
-    def __init__(self, transform=None, reverse_transform=None, is_valid=None):
+    def _run(self, function, table_data, reverse=False):
+        table_data = table_data.copy()
+        if self._columns:
+            if reverse:
+                columns = reversed(self._columns)
+            else:
+                columns = self._columns
+
+            for column in columns:
+                try:
+                    table_data = function(table_data, column)
+                except TypeError:
+                    table_data[column] = function(table_data[column])
+
+        else:
+            table_data = function(table_data)
+
+        return table_data
+
+    def _run_transform(self, table_data):
+        return self._run(self._transform, table_data)
+
+    def _run_reverse_transform(self, table_data):
+        return self._run(self._reverse_transform, table_data, reverse=True)
+
+    def _run_is_valid(self, table_data):
+        if self._columns:
+            try:
+                valid = [self._is_valid(table_data, column) for column in self._columns]
+            except TypeError:
+                valid = [self._is_valid(table_data[column]) for column in self._columns]
+
+            return np.logical_and.reduce(valid)
```

Comment:
Is this faster than `numpy.all`? I think that can also be used with multi-dimensional arrays